be/src/vec/olap/vgeneric_iterators.cpp

```diff
@@ -374,6 +374,11 @@ Status VUnionIterator::init(const StorageReadOptions& opts) {
         return Status::OK();
     }
 
+    // we use back() and pop_back() of std::vector to handle each iterator,
```

Comment:
we'd better to make iterator's order right outside of the UnionIterator?
It's tricky to do this kind of operation inside UnionIterator, if some other developers changed the iterator's order outside of UnionIterator for some reason, the result will not be wrong.
Consider to do this operation at `beta_rowset_reader.cpp:L218`, just before `  _iterator = vectorized::new_union_iterator(std::move(iterators));`, and we should add a DCHECK on `iter.data_id()` to check that the iterators order right.

Reply:
As I understand, the VUnionIterator's behaviour is to output data from the list of segment iterators, from begin to end.

The user of VUnionIterator is reponsible of the corrrect order of the list of segment iterators, and VUnionIterator should keep the order of output data is the same as the input order of list of segment iterators. 

As in the code comment, we use back() and pop_back() of std::vector to handle each iterator, so reverse the vector here to keep result block of next_batch to be in the same order as the original segments.

Reply:
My bad, I understand the context wrong at the beginning. I thought why you need to reverse the iterator here because the input is reversed, so I suggested that we should adjust the input.